crates/cxx-qt/include/thread.h

```diff
@@ -52,15 +54,13 @@ class CxxQtThread final
   }
 
   template<typename A>
-  void queue(::rust::Fn<void(T& self, ::rust::Box<A> arg)> func,
-             ::rust::Box<A> arg) const
+  ::std::uint8_t queue(::rust::Fn<void(T& self, ::rust::Box<A> arg)> func,
+                       ::rust::Box<A> arg) const
   {
     // Ensure that we can read the pointer and it's not being written to
     const auto guard = ::std::shared_lock(m_obj->mutex);
     if (!m_obj->ptr) {
-      throw ::std::runtime_error(
-        "Cannot queue function pointer as object has been destroyed");
-      return;
+      return 1;
```

Comment:
nit: Magic number.
We should probably create constants for these on the C++ side to mirror the Rust enum.

e.g.
```cpp
const static ::std::uint8_t ObjectDestroyed = 1;
const static ::std::uint8_t InvokeMethodFailed = 1;
```

Reply:
done